### ISSUE TITLE
Evaluation took much more time when enable eval_table_size 

### ISSUE THREAD

**Issue by LeDuySon** [axolotl-ai-cloud/axolotl#949]
### Please check that this issue hasn't been reported before.

- [X] I searched previous [Bug Reports](https://github.com/OpenAccess-AI-Collective/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

The evaluation time is expected to increase but not much like the current one

### Current behaviour

The evaluation time took a long time when enable eval_table_size to process only about 50 samples
![image](https://github.com/OpenAccess-AI-Collective/axolotl/assets/33374938/e9f3ac4c-4606-4b94-9949-939eb9ef32f2)

### Steps to reproduce

accelerate launch -m axolotl.cli.train configs/pygmalion2_qlora.yaml \
                                       --wandb_project ${wandb_project} \
                                       --wandb_run_id ${run_id} \
                                       --output_dir "${output_dir}/${run_id}" \
                                       --num_epochs 10 

### Config yaml

base_model: PygmalionAI/pygmalion-2-13b
model_type: LlamaForCausalLM
tokenizer_type: LlamaTokenizer
is_llama_derived_model: true

load_in_8bit: false
load_in_4bit: true
strict: false

datasets:
  - path: ./test
    type:
      system_prompt: "..."
      field_system: <|system|>
      format: "<|user|> {instruction} <|model|>"
      no_input_format: "<|user|> {instruction} <|model|>"

dataset_prepared_path:
val_set_size: 0.1
output_dir: ./outputs/pygmalion2_qlora

adapter: qlora
lora_model_dir:

sequence_len: 2048
sample_packing: false
pad_to_sequence_len: true

lora_r: 16
lora_alpha: 8
lora_dropout: 0.05
lora_target_modules:
  - q_proj
  - v_proj
  # - k_proj
  # - o_proj
lora_target_linear: true
lora_fan_in_fan_out:

wandb_project: pygmalion2
wandb_entity:
wandb_watch:
wandb_run_id:
wandb_log_model:
# wandb eval table
eval_table_size: 100 # Approximate number of predictions sent to wandb depending on batch size. Enabled above 0. Default is 0
eval_table_max_new_tokens: 1024 # Total number of tokens generated for predictions sent to wandb. Default is 128

gradient_accumulation_steps: 1
micro_batch_size: 16
num_epochs: 10
optimizer: paged_adamw_32bit
lr_scheduler: cosine
learning_rate: 0.0002

train_on_inputs: false
group_by_length: false
bf16: true
fp16: false
tf32: false

gradient_checkpointing: true
early_stopping_patience:
resume_from_checkpoint:
local_rank:
logging_steps: 1
xformers_attention:
flash_attention: true

warmup_steps: 10
eval_steps: 0.05
eval_batch_size: 8
eval_sample_packing: false
save_steps: 
debug:
deepspeed:
weight_decay: 0.0
fsdp:
fsdp_config:
special_tokens:
  bos_token: "<s>"
  eos_token: "</s>"
  unk_token: "<unk>"

seed: 1

### Which Operating Systems are you using?

- [X] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.10

### axolotl branch-commit

738a0576743e40a88ca29075bd2bf4ff8c6e6fc4

### Acknowledgements

- [X] My issue title is concise, descriptive, and in title casing.
- [X] I have searched the existing issues to make sure this bug has not been reported yet.
- [X] I am using the latest version of axolotl.
- [X] I have provided enough information for the maintainers to reproduce and diagnose the issue.

**Comment by NanoCode012** [comment]
Hi, sorry for late reply. `eval_table_size` is not working as well as expected unfortunately.  That's why it's disabled by default.

**Comment by ysawej** [comment]
I have been looking to get eval outputs too and I ran into the same issue. It seems to go into model.forward and stop. This is only a problem when I run it with accelerate, and its hard to debug whats not working.

**Comment by NanoCode012** [comment]
@ysawej . Sorry for that. It seems that this feature is not properly maintained anymore. If you would like, would you be able to take a look and perhaps submit a PR fix?
